addons/mail/static/src/messaging/entity/entity_core.js

```diff
@@ -0,0 +1,246 @@
+odoo.define('mail.messaging.entity.core', function (require) {
+'use strict';
+
+const { patchClassMethods, patchInstanceMethods } = require('mail.messaging.utils');
+
+const registry = {};
+
+//------------------------------------------------------------------------------
+// Private
+//------------------------------------------------------------------------------
+
+/**
+ * @private
+ * @param {string} entityName
+ * @returns {Object}
+ */
+function _getEntryFromEntityName(entityName) {
+    if (!registry[entityName]) {
+        registry[entityName] = {
+            dependencies: [],
+            factory: undefined,
+            name: entityName,
+            patches: [],
+        };
+    }
+    return registry[entityName];
+}
+
+/**
+ * @private
+ * @param {string} entityName
+ * @param {string} patchName
+ * @param {Object} patch
+ * @param {Object} [param3={}]
+ * @param {Object} [param3.isOnClass=false]
+ */
+function _registerPatchEntity(entityName, patchName, patch, { isOnClass = false } = {}) {
+    const entry = _getEntryFromEntityName(entityName);
+    Object.assign(entry, {
+        patches: (entry.patches || []).concat([{
+            isOnClass,
+            name: patchName,
+            patch,
+        }]),
+    });
+}
+
+//------------------------------------------------------------------------------
+// Public
+//------------------------------------------------------------------------------
+
+/**
+ * @param {Object} Entities
+ * @throws {Error} in case some relations are not correct
+ */
```

Comment:
Should document just a bit the goal of the method (but it is pretty obvious reading its code).